runtime/jcl/common/reflecthelp.c

```diff
@@ -833,9 +833,17 @@ createField(struct J9VMThread *vmThread, jfieldID fieldID)
 #if JAVA_SPEC_VERSION >= 15
 	/* trust that static final fields and final record or hidden class fields will not be modified. */
 	if (J9_ARE_ALL_BITS_SET(j9FieldID->field->modifiers, J9AccFinal)) {
+#if JAVA_SPEC_VERSION >= 16
+		J9UTF8 *superClassName = J9ROMCLASS_SUPERCLASSNAME(j9FieldID->declaringClass->romClass);
+#endif /* JAVA_SPEC_VERSION >= 16 */
+
 		if (J9_ARE_ALL_BITS_SET(j9FieldID->field->modifiers, J9AccStatic)
-			|| J9ROMCLASS_IS_RECORD(j9FieldID->declaringClass->romClass)
 			|| J9ROMCLASS_IS_HIDDEN(j9FieldID->declaringClass->romClass)
+			|| (J9ROMCLASS_IS_RECORD(j9FieldID->declaringClass->romClass)
+#if JAVA_SPEC_VERSION >= 16
+				&& (0 == memcmp("java/lang/Record", J9UTF8_DATA(superClassName), J9UTF8_LENGTH(superClassName)))
```

Comment:
This should use the J9UTF8_LITERAL_EQUALS macro - as the code stand it could read undefined memory (memcmp does not stop at the string terminator).

Reply:
The check on superclass has been removed in this file as suggested below.